### ISSUE TITLE
Different computers generate different citation keys for same items in shared library

### ISSUE THREAD

**Issue by samuelmehr** [retorquere/zotero-better-bibtex#3019]
### Debug log ID

6FSQG35P-refs-apse/6.7.240-7

### What happened?

Hi, we use Zotero+BBT with R Markdown to write manuscripts in my lab and have run into an issue where Zotero instances from different users, and also the same user with multiple instances of Zotero (e.g. on their laptop and on their desktop), generate different citation keys for items in the same shared library.

For example, I just tested this with two machines using the same shared library. Both machines are using `auth+year` citation keys, but three items on one computer have the keys `Mehr2017`, `Mehr2017a`, `Mehr2017b`, respectively, and on the other, they have `Mehr2017a`, `Mehr2017b`, `Mehr2017c`. Both machines have "Keep keys unique within each library" selected.

We tried playing with different auth + year formulae (like auth.fold + year or auth.capitalize + year) and this didn't seem to affect the citation keys.

This is an issue because on a collaborative R Markdown manuscript, multiple users may generate new .bib files and if their citation keys are inconsistent, either an existing citation key will fail to knit properly, or worse, it will knit with the wrong item cited (which we might not notice in the reflist).

Are we doing something wrong here?

**Comment by retorquere** [comment]
No, until Zotero adds a citation key field citation keys are local to the installation of Zotero they're generated on; if you want syncable keys, right now your only option is to enable [auto-pin](https://retorque.re/zotero-better-bibtex/installation/preferences/index.html#automatically-pin-citation-key-after), which will write the keys into the `extra` field, and that syncs. BBT always prefers the key in the `extra` field.

**Comment by retorquere** [comment]
does this address your question?
